### PR TITLE
Improve SCIM1 to return time in UTC format for user creation response.

### PR DESCRIPTION
**Pull request by dewniMW**
This PR resolves https://github.com/wso2/product-is/issues/6718

### CODE REVIEW

File: modules/charon-core/src/main/java/org/wso2/charon/core/schema/SCIMConstants.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
     public static final String APPLICATION_JSON = "application/json";
     public static final String APPLICATION_XML = "application/xml";
 
-    public static final String dateTimeFormat = "yyyy-MM-dd'T'HH:mm:ss";
```
Comment (DMHP) [inline modules/charon-core/src/main/java/org/wso2/charon/core/schema/SCIMConstants.java]: Please don't remove the public variable. Mark it as deprecated.

**Comment by dewniMW** [comment]
I ran the integration tests in master and the tests passed successfully.

File: modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java [hunk 2]
```diff
@@ -110,9 +111,21 @@ public static Date parseDateTime(String dateTimeString) throws CharonException {
...
-        String formattedDate = sdf.format(date);
-        return formattedDate;
+
+        SimpleDateFormat sdf = new SimpleDateFormat(SCIMConstants.dateTimeFormat2);
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+        return sdf.format(date);
+    }
+
+    /*
+     * SCIM spec requires date time to be adhered to XML Schema Datatypes Specification
```
Comment (somindatommy) [inline modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: Complete the method comment with `.` and param descriptions.

File: modules/charon-core/src/test/java/org/wso2/charon/core/util/AttributeUtilTest.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ public void testGetAttributeURI() throws Exception {
                 { false, SCIMSchemaDefinitions.DataType.BOOLEAN, "false" },
                 { new Byte((byte) 1), SCIMSchemaDefinitions.DataType.BINARY, "1" },
                 { -1.1, SCIMSchemaDefinitions.DataType.DECIMAL, "-1.1" },
-                { getTestTime(), SCIMSchemaDefinitions.DataType.DATE_TIME, "2017-10-18T18:00:00" } };
+                { getTestTime(), SCIMSchemaDefinitions.DataType.DATE_TIME, "2017-10-18T18:00:00Z" } };
```
Comment (somindatommy) [inline modules/charon-core/src/test/java/org/wso2/charon/core/util/AttributeUtilTest.java]: Formatting Error. Extra white spaces.

File: modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java [hunk 4]
```diff
@@ -110,9 +111,21 @@ public static Date parseDateTime(String dateTimeString) throws CharonException {
...
+     *
+     * @param date
+     */
+    public static String formatDateTimeLocal(Date date) {
+
```
Comment (somindatommy) [inline modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: What happens when the date is Null?

Comment (somindatommy) [inline modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: What happens when the date is NULL?

File: modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java [hunk 5]
```diff
@@ -110,9 +111,21 @@ public static Date parseDateTime(String dateTimeString) throws CharonException {
      * @param date
      */
     public static String formatDateTime(Date date) {
-        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
-        String formattedDate = sdf.format(date);
-        return formattedDate;
+
+        SimpleDateFormat sdf = new SimpleDateFormat(SCIMConstants.dateTimeFormat2);
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
```
Comment (somindatommy) [inline modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: Shall we have a constant for this? WDYT?

Comment (dewniMW) [reply modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: fixed: https://github.com/wso2/charon/pull/270

File: modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java (hunk 2 above)
Comment (dewniMW) [reply modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: fixed https://github.com/wso2/charon/pull/270

File: modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java (hunk 4 above)
Comment (dewniMW) [reply modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: fixed https://github.com/wso2/charon/pull/270

Comment (dewniMW) [reply modules/charon-core/src/main/java/org/wso2/charon/core/util/AttributeUtil.java]: fixed: https://github.com/wso2/charon/pull/270

File: modules/charon-core/src/test/java/org/wso2/charon/core/util/AttributeUtilTest.java (hunk 3 above)
Comment (dewniMW) [reply modules/charon-core/src/test/java/org/wso2/charon/core/util/AttributeUtilTest.java]: fixed: https://github.com/wso2/charon/pull/270

**Pull request merged**